src/braket/experimental/autoqasm/program/program.py

```diff
@@ -404,14 +423,31 @@ def get_expression_var(self, expression: FreeParameterExpression) -> oqpy.FloatV
         self.get_oqpy_program().declare(var)
         return var
 
-    def get_free_parameters(self) -> list[oqpy.FloatVar]:
+    def get_free_parameters(self) -> list[oqpy.Var]:
         """Return a list of named oqpy.Vars that are used as free parameters in the program."""
         return list(self._free_parameters.values())
 
+    def get_free_parameter(self, name: str) -> oqpy.Var | None:
+        """Return the oqpy.Var associated with the variable name `name` in the program."""
+        return self._free_parameters.get(name, None)
+
+    def get_output_parameter(self, name: str) -> oqpy.Var | None:
+        """Return the oqpy.Var associated with the output variable `name` in the program."""
+        return self._output_parameters.get(name, None)
+
     def add_io_declarations(self) -> None:
         """Add input and output declaration statements to the program."""
         root_oqpy_program = self.get_oqpy_program(scope=ProgramScope.MAIN)
-        for parameter in self.get_free_parameters():
+        for parameter_name, parameter in self._free_parameters.items():
+            # Sometimes the names get overwritten -- see aq.types
```

Comment:
what does "see aq.types" mean? I looked but couldn't find anything obvious.

Reply:
Ah, I think I had added a TODO and then removed it. I'll update this comment. Thanks!